app/controllers/api/v1/admin/domain_blocks_controller.rb

```diff
@@ -19,12 +19,10 @@ class Api::V1::Admin::DomainBlocksController < Api::BaseController
   def create
     authorize :domain_block, :create?
 
-    existing_domain_block = resource_params[:domain].present? ? DomainBlock.rule_for(resource_params[:domain]) : nil
-    return render json: existing_domain_block, serializer: REST::Admin::ExistingDomainBlockErrorSerializer, status: 422 if existing_domain_block.present?
-
```

Comment:
Removing this introduces an inconsistency between the web-based admin interface and the API, where in the former, creating a domain block for a sub-domain will error out, and in the latter, creating such a domain block will succeed.

Overall, it may make sense to allow the latter, but `UnblockDomainService` currently assumes that when lifting a domain block, no other domain block will apply, so before allowing this change we need to take care of that.